crates/protocol/src/batch_tx.rs

```diff
@@ -29,10 +29,34 @@ impl BatchTransaction {
 
 impl From<&BatchTransaction> for Bytes {
```

Comment:
I personally very much dislike these kinds of conversions, this should just be a function instead.

this way you get proper intellisense, docs and don't have to do `(&v).into()`